### ISSUE TITLE
UI is hard to use when the user is moving the mouse while clicking

### ISSUE THREAD

**Issue by Tantacrul** [musescore/MuseScore#16012]
**Describe the bug**
See this video uploaded by a user who is unable to select many items on the interface: https://www.youtube.com/watch?v=KiGPBrbKeN0

I'm guessing there is some issue where mouse clicks are being cancelled when the cursor is being moves (even a little bit).

I'm not sure what the solution is:
- Trigger UI events on MouseUp, rather than MouseDown?
- Ignore cursor movement when clicking on UI elements?
- Establish a permitted amount of cursor drift when clicking?

**Expected behavior**
This user should be having none of these problems.

**Comment by cbjeukendrup** [comment]
I found the reason; it's relatively easy to solve, fortunately.

We do most things "on click" (rather than "mouse up" or "mouse down"). From the QML docs:
> A click is defined as a press followed by a release, both inside the [MouseArea](https://doc.qt.io/qt-5/qml-qtquick-mousearea.html) (pressing, moving outside the [MouseArea](https://doc.qt.io/qt-5/qml-qtquick-mousearea.html), and then moving back inside and releasing is also considered a click).

So it should work... but it doesn't.

And that turns out to be because we have attached a signal handler to the `pressAndHold` signal. In this handler, we simply forward the signal by calling another signal. This forwarded signal is then used in a few places throughout the app. But the handler where this happens, is thus connected to the `pressAndHold` signal for every FlatButton.
When QML detects that something is connected to this signal, the button starts apparently behaving differently and the click signal doesn't work anymore as described; it only works if you keep the mouse still. I'm inclined to call this a QML bug, but that doesn't really bring us further...
So to fix it for 99,9% of all buttons*, we should simply make sure that nothing is connected to that `pressAndHold` signal. The easiest way to do it, is by removing that "forwarded" signal, and in places where it was used simply connect to `mouseArea.pressAndHold` directly. A slightly more advanced way would be to put the "forwarder" handler in a `Connections` object that is only `enabled` when something is connected to the forwarded signal. It's currently not possible to check that in QML directly, so we'll have to write a small C++ utility function to achieve that. Not sure whether that's worth the effort.

\* Namely those where we don't have any functionality on "pressAndHold"


Alternatively, we could try if something like TapHandler works better than MouseArea, but I have a gut feeling that it will be only worse.
